core/src/test/java/org/mule/tck/util/TestTimeSupplier.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.tck.util;
+
+import java.util.concurrent.TimeUnit;
+import java.util.function.Supplier;
+
+/**
+ * A test {@link Supplier} for externalizing the system time. It is initialised at a given
+ * initial {@link #timeInMillis} using the {@link #TestTimeSupplier(long)} constructor. The value can be
+ * retrieved through the {@link #get()} method.
+ * <p/>
+ * The usefulness of ths class comes from the behavior provided by the {@link #move(long, TimeUnit)}
+ * method which allows changing that value per the tests needs. Again, the modified value is to be retrieved
```

Comment:
changed it to be more clear. What I meant is that after calling move(), you don't get to retrieve the original value again